Add the browser `next`

nlewo opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Add the browser [`next`](https://github.com/atlas-engineer/next).
I also had to add several Lisp libraries which are dependencies of `next`.

Thanks @Ambrevar for his help!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @7c6f434c

nlewo committed:
lispPackages.iolib: patch lisp files to use the libfixposix library

nlewo committed:
lispPackages: add 11 packages

The following packages has been added:
- cl-css
- dbus
- log4cl
- mk-string-metrics
- prove-asdf
- sqlite
- str
- swank
- trivia
- trivial-clipboard
- unix-opts

nlewo committed:
next: init at 1.3.1

7c6f434c commented:
It looks like you are adding fixes before adding the corresponding packages… It also looks like some of the generated files (like `dbus.nix`) are missing. And I think in general it is better to have `sqlite` fix in the same commit as addition (adding libraries and in the same commit adding something to help them find native dependencies is a normal workflow).

jonringer commented on pkgs/applications/networking/browsers/next/default.nix:
```diff
@@ -0,0 +1,106 @@
+    sha256 = "01fn1f080ydk0wj1bwkyakqz93bdq9xb5x8qz820jpl9id17bqgj";
+  };
+
+  next-gtk-webkit = stdenv.mkDerivation rec {
```
i feel like this should be broken out into it's own file/package. you could still do:
```
inherit (next) src version;
```

nlewo commented:
@7c6f434c oups, sorry! I added missing files and reworked the commit flow. 
I patch the package `iolib` which was already packaged. So, i prefer to add this patch in a separated commit.
I thought the `sqlite` package was also already existing, which is actually not the case. I squashed this commit into the main `lispPacakges` commit.

yurrriq commented on pkgs/development/lisp-modules/quicklisp-to-nix-output/chunga.nix:
```diff
@@ -3,7 +3,7 @@ args @ { fetchurl, ... }:
   version = ''20180131-git'';
 
-  description = '''';
+  description = ''System lacks description'';
```
I'm guessing this was generated. But maybe we could back out this and similar changes so the diff is smaller.

nlewo replied:
Since this has been generated, I would prefer to not edit it manually.

7c6f434c commented:
@GrahamcOfBorg build next

nlewo commented:
@GrahamcOfBorg build next

7c6f434c commented:
@GrahamcOfBorg build next

matthuszagh commented:
Next only starts for me if I run it with `$ export $(dbus-launch) && next`. Additionally, when launched I only get blank windows.

7c6f434c commented:
>Next only starts for me if I run it with `$ export $(dbus-launch) && next`. Additionally, when launched I only get blank windows.

The first part happens, I guess many people have a DBus session in a typical situation — we could wrap it like LibreOffice is to auto-launch a DBus session unless present. The second is more interesting question…

Ambrevar commented:
Note that Next 1.3.2 (to be released this week?) will auto-launch D-Bus.
You can configure the D-Bus path by changing the value of +dbus-launch-command+
in global.lisp.

For the blank window: is glib-networking and gsettings-desktop-schemas
in the user profile?  Those are requirements (propagated inputs) for
WebKitGTK browsers.

matthuszagh commented:
Unfortunately I'm still having trouble with a blank window. I tried adding `glib-networking` to the next-gtk-webkit nix file under `propagatedBuildInputs` and moving `gsettings-desktop-schema` to be propagated as well. I also tried just installing both of these packages in my user environment.

Ambrevar commented:
Do you see a minibuffer if you press `C-l`?

What output do you get in the shell?
Anything in ~/.local/share/next/next-gtk-webkit.log

matthuszagh commented:
C-l appears to work (shows recent websites) and it looks like it's able to open the websites fine too, but the problem looks like it's just with the display. Here's the full log from startup to navigating to github

```
** Message: 08:27:52.315: Starting platform port
** Message: 08:27:52.378: Method name: window_make
** Message: 08:27:52.378: Method parameter(s): 1
** Message: 08:27:52.409: Method result(s): window id 1
** Message: 08:27:52.418: Method name: buffer_make
** Message: 08:27:52.418: Method parameter(s): buffer ID 1, options {'cookies-path': '/home/matt/.local/share/next/cookies.txt'}
** Message: 08:27:52.422: Method result(s): buffer id 1
** Message: 08:27:52.430: Method name: window_active
** Message: 08:27:52.430: Method parameter(s): <no active window>
** Message: 08:27:52.433: Method name: window_active
** Message: 08:27:52.433: Method parameter(s): <no active window>
** Message: 08:27:52.442: Method name: minibuffer_evaluate_javascript
** Message: 08:27:52.442: Method parameter(s): window id 1, javascript (...)
** Message: 08:27:52.442: Method result(s): callback id 0
** Message: 08:27:52.445: Method name: window_set_minibuffer_height
** Message: 08:27:52.445: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:27:52.445: Method result(s): minibuffer preferred height 0
** Message: 08:27:52.448: Method name: window_active
** Message: 08:27:52.448: Method parameter(s): 1
** Message: 08:27:52.457: Method name: minibuffer_evaluate_javascript
** Message: 08:27:52.457: Method parameter(s): window id 1, javascript (...)
** Message: 08:27:52.457: Method result(s): callback id 1
** Message: 08:27:52.463: Method name: window_active
** Message: 08:27:52.463: Method parameter(s): 1
** Message: 08:27:52.466: Method name: window_active
** Message: 08:27:52.466: Method parameter(s): 1
** Message: 08:27:52.469: Method name: minibuffer_evaluate_javascript
** Message: 08:27:52.469: Method parameter(s): window id 1, javascript (...)
** Message: 08:27:52.469: Method result(s): callback id 2
** Message: 08:27:52.472: Method name: window_set_minibuffer_height
** Message: 08:27:52.472: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:27:52.472: Method result(s): minibuffer preferred height 0
** Message: 08:27:52.476: Method name: window_active
** Message: 08:27:52.476: Method parameter(s): 1
** Message: 08:27:52.479: Method name: minibuffer_evaluate_javascript
** Message: 08:27:52.479: Method parameter(s): window id 1, javascript (...)
** Message: 08:27:52.479: Method result(s): callback id 3
** Message: 08:27:52.483: Method name: buffer_evaluate_javascript
** Message: 08:27:52.483: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:27:52.483: Method result(s): callback id 0
** Message: 08:27:52.486: Method name: window_active
** Message: 08:27:52.486: Method parameter(s): 1
** Message: 08:27:52.489: Method name: window_set_active_buffer
** Message: 08:27:52.489: Method parameter(s): window id 1, buffer id 1
** Message: 08:27:52.489: Window 1 switches from buffer (null) to 1 with URI (null)
** Message: 08:27:52.492: Method name: window_active
** Message: 08:27:52.492: Method parameter(s): 1
** Message: 08:27:52.495: Method name: buffer_evaluate_javascript
** Message: 08:27:52.495: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:27:52.495: Method result(s): callback id 1
** Message: 08:27:52.498: Method name: window_set_active_buffer
** Message: 08:27:52.498: Method parameter(s): window id 1, buffer id 1
** Message: 08:27:52.498: Window 1 switches from buffer 1 to 1 with URI (null)
** Message: 08:27:52.501: Method name: window_active
** Message: 08:27:52.501: Method parameter(s): 1
** Message: 08:27:52.504: Method name: buffer_evaluate_javascript
** Message: 08:27:52.504: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:27:52.504: Method result(s): callback id 2
** Message: 08:27:52.508: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 0)
** Message: 08:27:52.508: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 1)
** Message: 08:27:52.508: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 2)
** Message: 08:27:52.508: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 3)
Cannot create EGL window surface: EGL_BAD_MATCH
** Message: 08:27:52.543: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 0)
** Message: 08:27:52.584: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 1)
** Message: 08:27:52.584: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 2)
** Message: 08:27:52.589: Method name: window_set_title
** Message: 08:27:52.589: Method parameter(s): 1, Next - *Help*
** Message: 08:27:52.593: Method name: window_set_title
** Message: 08:27:52.593: Method parameter(s): 1, Next - *Help*
** Message: 08:28:26.278: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (36, 'RETURN', ['R'], -1.0, -1.0, 65293, '1')
** Message: 08:28:26.281: Method name: generate_input_event
** Message: 08:28:26.281: Method parameter(s): window id '1', hardware_keycode 36, keyval 65293, modifiers 0
** Message: 08:32:23.548: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (46, 'l', ['C'], -1.0, -1.0, 108, '1')
** Message: 08:32:23.553: Method name: window_active
** Message: 08:32:23.553: Method parameter(s): 1
** Message: 08:32:23.556: Method name: window_active
** Message: 08:32:23.556: Method parameter(s): 1
** Message: 08:32:23.559: Method name: minibuffer_evaluate_javascript
** Message: 08:32:23.559: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:23.559: Method result(s): callback id 4
** Message: 08:32:23.561: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 4)
** Message: 08:32:23.563: Method name: window_active
** Message: 08:32:23.563: Method parameter(s): 1
** Message: 08:32:23.566: Method name: minibuffer_evaluate_javascript
** Message: 08:32:23.566: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:23.566: Method result(s): callback id 5
** Message: 08:32:23.569: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 5)
** Message: 08:32:23.570: Method name: window_active
** Message: 08:32:23.570: Method parameter(s): 1
** Message: 08:32:23.573: Method name: window_active
** Message: 08:32:23.573: Method parameter(s): 1
** Message: 08:32:23.582: Method name: window_active
** Message: 08:32:23.582: Method parameter(s): 1
** Message: 08:32:23.585: Method name: minibuffer_evaluate_javascript
** Message: 08:32:23.585: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:23.585: Method result(s): callback id 6
** Message: 08:32:23.586: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 6)
** Message: 08:32:23.588: Method name: window_active
** Message: 08:32:23.588: Method parameter(s): 1
** Message: 08:32:23.591: Method name: window_set_minibuffer_height
** Message: 08:32:23.591: Method parameter(s): window id 1, minibuffer height 200
** Message: 08:32:23.591: Method result(s): minibuffer preferred height 0
** Message: 08:32:23.624: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (46, 'l', ['C', 'R'], -1.0, -1.0, 108, '1')
** Message: 08:32:25.639: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (36, 'RETURN', @as [], -1.0, -1.0, 65293, '1')
** Message: 08:32:25.642: Method name: window_active
** Message: 08:32:25.642: Method parameter(s): 1
** Message: 08:32:25.645: Method name: window_active
** Message: 08:32:25.645: Method parameter(s): 1
** Message: 08:32:25.648: Method name: window_active
** Message: 08:32:25.648: Method parameter(s): 1
** Message: 08:32:25.651: Method name: minibuffer_evaluate_javascript
** Message: 08:32:25.651: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:25.651: Method result(s): callback id 7
** Message: 08:32:25.652: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 7)
** Message: 08:32:25.654: Method name: window_set_minibuffer_height
** Message: 08:32:25.654: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:25.654: Method result(s): minibuffer preferred height 0
** Message: 08:32:25.657: Method name: window_active
** Message: 08:32:25.657: Method parameter(s): 1
** Message: 08:32:25.661: Method name: minibuffer_evaluate_javascript
** Message: 08:32:25.661: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:25.661: Method result(s): callback id 8
** Message: 08:32:25.661: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 8)
** Message: 08:32:25.664: Method name: window_set_minibuffer_height
** Message: 08:32:25.664: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:25.664: Method result(s): minibuffer preferred height 0
** Message: 08:32:25.667: Method name: window_active
** Message: 08:32:25.667: Method parameter(s): 1
** Message: 08:32:25.670: Method name: window_active
** Message: 08:32:25.670: Method parameter(s): 1
** Message: 08:32:25.688: Method name: window_active
** Message: 08:32:25.688: Method parameter(s): 1
** Message: 08:32:25.691: Method name: window_active
** Message: 08:32:25.691: Method parameter(s): 1
** Message: 08:32:25.729: Method name: buffer_evaluate_javascript
** Message: 08:32:25.729: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:25.729: Method result(s): callback id 3
** Message: 08:32:25.731: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 3)
** Message: 08:32:25.731: RPC message: request_resource (buffer id, URI, cookies, event_type, is_new_window, is_known_type, button, modifiers) = ('1', 'https://github.com/', '_octo=GH1.1.53549842.1568309574; logged_in=no; _ga=GA1.2.262583195.1568309575', 'other', false, true, 'button0', @as [])
** Message: 08:32:25.743: RPC message: push_input_event (keycode, keystring, modifiers, x, y, low level data, window id) = (36, 'RETURN', ['R'], -1.0, -1.0, 65293, '1')
** Message: 08:32:25.746: Method name: generate_input_event
** Message: 08:32:25.746: Method parameter(s): window id '1', hardware_keycode 36, keyval 65293, modifiers 0
** Message: 08:32:26.089: RPC message: request_resource (buffer id, URI, cookies, event_type, is_new_window, is_known_type, button, modifiers) = ('1', 'https://github.com/', '_octo=GH1.1.53549842.1568309574; logged_in=no; _ga=GA1.2.262583195.1568309575; has_recent_activity=1; _gh_sess=MXpOL2lNc0djTW1qRThLY1Bhd3JJdzdabkEraDBoL2VFZnBqWDVXczFyUGt5TTBJTzcrUnBZcmcwcXBFSVFoTklrbFpQVjVnd1RpSmdDOVRXNUFDaDZPa1BlUlVmdE40N3JsL0ovc3RaT0Nkb2dlR0pFS2N0TmxBR1ZIZXQ3THFoTTBmbzFrU2YxcmR4YVdmb3gzU2hRPT0tLXRSNW5jSjF6dXorUHo5YUtZeTU4aGc9PQ%3D%3D--db7e7c13e8d4181da48d2ecdc5332cc76b3f3a50', 'other', false, true, '', @as [])
** Message: 08:32:26.091: RPC message: buffer_did_commit_navigation ('1', 'https://github.com/')
** Message: 08:32:26.091: RPC message: buffer_did_commit_navigation ('1', 'https://github.com/')
** Message: 08:32:26.094: Method name: window_active
** Message: 08:32:26.094: Method parameter(s): 1
** Message: 08:32:26.097: Method name: buffer_evaluate_javascript
** Message: 08:32:26.097: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.097: Method result(s): callback id 4
** Message: 08:32:26.100: Method name: window_active
** Message: 08:32:26.100: Method parameter(s): 1
** Message: 08:32:26.103: Method name: window_active
** Message: 08:32:26.103: Method parameter(s): 1
** Message: 08:32:26.106: Method name: window_active
** Message: 08:32:26.106: Method parameter(s): 1
** Message: 08:32:26.109: Method name: buffer_evaluate_javascript
** Message: 08:32:26.109: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.109: Method result(s): callback id 5
** Message: 08:32:26.110: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 4)
** Message: 08:32:26.110: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 5)
** Message: 08:32:26.113: Method name: window_active
** Message: 08:32:26.113: Method parameter(s): 1
** Message: 08:32:26.116: Method name: window_active
** Message: 08:32:26.116: Method parameter(s): 1
** Message: 08:32:26.141: Method name: window_active
** Message: 08:32:26.141: Method parameter(s): 1
** Message: 08:32:26.144: Method name: window_active
** Message: 08:32:26.144: Method parameter(s): 1
** Message: 08:32:26.147: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.147: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.147: Method result(s): callback id 9
** Message: 08:32:26.148: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 9)
** Message: 08:32:26.150: Method name: window_set_minibuffer_height
** Message: 08:32:26.151: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.151: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.153: Method name: window_active
** Message: 08:32:26.153: Method parameter(s): 1
** Message: 08:32:26.156: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.157: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.157: Method result(s): callback id 10
** Message: 08:32:26.157: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 10)
** Message: 08:32:26.159: Method name: window_active
** Message: 08:32:26.159: Method parameter(s): 1
** Message: 08:32:26.162: Method name: buffer_evaluate_javascript
** Message: 08:32:26.162: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.162: Method result(s): callback id 6
** Message: 08:32:26.165: Method name: window_active
** Message: 08:32:26.165: Method parameter(s): 1
** Message: 08:32:26.166: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 6)
** Message: 08:32:26.168: Method name: window_active
** Message: 08:32:26.168: Method parameter(s): 1
** Message: 08:32:26.171: Method name: window_active
** Message: 08:32:26.171: Method parameter(s): 1
** Message: 08:32:26.174: Method name: buffer_evaluate_javascript
** Message: 08:32:26.174: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.174: Method result(s): callback id 7
** Message: 08:32:26.174: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 7)
** Message: 08:32:26.177: Method name: window_active
** Message: 08:32:26.177: Method parameter(s): 1
** Message: 08:32:26.180: Method name: window_active
** Message: 08:32:26.180: Method parameter(s): 1
** Message: 08:32:26.183: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.183: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.183: Method result(s): callback id 11
** Message: 08:32:26.183: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 11)
** Message: 08:32:26.186: Method name: window_set_minibuffer_height
** Message: 08:32:26.186: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.186: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.189: Method name: window_active
** Message: 08:32:26.189: Method parameter(s): 1
** Message: 08:32:26.193: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.193: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.193: Method result(s): callback id 12
** Message: 08:32:26.193: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 12)
** Message: 08:32:26.197: Method name: window_set_title
** Message: 08:32:26.197: Method parameter(s): 1, Next - The world’s leading software development platform · GitHub - https://github.com/
** Message: 08:32:26.200: Method name: window_set_title
** Message: 08:32:26.201: Method parameter(s): 1, Next - The world’s leading software development platform · GitHub - https://github.com/
** Message: 08:32:26.843: RPC message: buffer_did_finish_navigation ('1', 'https://github.com/')
** Message: 08:32:26.847: Method name: window_active
** Message: 08:32:26.847: Method parameter(s): 1
** Message: 08:32:26.849: Method name: window_active
** Message: 08:32:26.849: Method parameter(s): 1
** Message: 08:32:26.853: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.853: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.853: Method result(s): callback id 13
** Message: 08:32:26.854: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 13)
** Message: 08:32:26.856: Method name: window_set_minibuffer_height
** Message: 08:32:26.856: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.856: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.864: Method name: window_active
** Message: 08:32:26.864: Method parameter(s): 1
** Message: 08:32:26.867: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.867: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.867: Method result(s): callback id 14
** Message: 08:32:26.868: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 14)
** Message: 08:32:26.870: Method name: window_active
** Message: 08:32:26.870: Method parameter(s): 1
** Message: 08:32:26.873: Method name: window_active
** Message: 08:32:26.873: Method parameter(s): 1
** Message: 08:32:26.877: Method name: window_active
** Message: 08:32:26.877: Method parameter(s): 1
** Message: 08:32:26.881: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.881: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.881: Method result(s): callback id 15
** Message: 08:32:26.882: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 15)
** Message: 08:32:26.884: Method name: window_set_minibuffer_height
** Message: 08:32:26.884: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.884: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.887: Method name: window_active
** Message: 08:32:26.887: Method parameter(s): 1
** Message: 08:32:26.890: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.890: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.890: Method result(s): callback id 16
** Message: 08:32:26.891: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 16)
** Message: 08:32:26.894: Method name: window_set_minibuffer_height
** Message: 08:32:26.894: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.894: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.897: Method name: window_active
** Message: 08:32:26.897: Method parameter(s): 1
** Message: 08:32:26.900: Method name: buffer_evaluate_javascript
** Message: 08:32:26.900: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.900: Method result(s): callback id 8
** Message: 08:32:26.900: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 8)
** Message: 08:32:26.904: Method name: window_active
** Message: 08:32:26.904: Method parameter(s): 1
** Message: 08:32:26.907: Method name: buffer_evaluate_javascript
** Message: 08:32:26.907: Method parameter(s): buffer id 1, javascript (...)
** Message: 08:32:26.907: Method result(s): callback id 9
** Message: 08:32:26.907: RPC message: buffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 9)
** Message: 08:32:26.910: Method name: window_active
** Message: 08:32:26.910: Method parameter(s): 1
** Message: 08:32:26.913: Method name: window_active
** Message: 08:32:26.913: Method parameter(s): 1
** Message: 08:32:26.916: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.916: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.916: Method result(s): callback id 17
** Message: 08:32:26.917: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 17)
** Message: 08:32:26.919: Method name: window_set_minibuffer_height
** Message: 08:32:26.919: Method parameter(s): window id 1, minibuffer height 25
** Message: 08:32:26.919: Method result(s): minibuffer preferred height 0
** Message: 08:32:26.922: Method name: window_active
** Message: 08:32:26.922: Method parameter(s): 1
** Message: 08:32:26.925: Method name: minibuffer_evaluate_javascript
** Message: 08:32:26.925: Method parameter(s): window id 1, javascript (...)
** Message: 08:32:26.925: Method result(s): callback id 18
** Message: 08:32:26.926: RPC message: minibuffer_javascript_call_back (buffer id, javascript, callback id) = (1, ..., 18)
```
Here's also a screenshot of window (top half is cut off since its all white anyway)
![next](https://user-images.githubusercontent.com/7377393/65259274-46044280-dab9-11e9-899d-393a059874eb.png)

Note also that I haven't yet created an `init.lisp` file. I didn't see anywhere that that was necessary to get it running, but maybe I missed something.

Ambrevar commented:
No you don't need init.lisp.

If you can display the minibuffer, then webkitgtk is working (the
minibuffer is a webview).

Just got a hunch: try browsing http://imdb.com (not HTTPS).
I can think of 2 possible issues:

- The 'Cannot create EGL window surface: EGL_BAD_MATCH' hints that
  hardware acceleration is failing on your hardware.  You'll probably
  have the same issue with other browsers like Epiphany.  Which video
  driver do you have?

- An HTTPS issue (although this should print "Unacceptable TLS certificate).

matthuszagh commented:
imdb doesn't work either. I'm using the nvidia proprietary drivers (v435.21)

Ambrevar commented:
Does any other WebKitGTK browser work for you?

matthuszagh commented:
Epiphany works fine.

Ambrevar commented:
Hmmm, I'm running out of ideas.
Can anyone else reproduce?

What if you build and run Next locally?

nlewo commented:
@matthuszagh I'm not able to reproduce your blank page issue :(
Are you able to test this package from another system/machine?

@GrahamcOfBorg build next

matthuszagh commented:
@Ambrevar are you saying to build it from source outside of Nix tooling? I can try although I know nothing about quicklisp so I'm not sure I'll be able to. And unfortunately I don't have another Nix machine yet... I'll keep digging and see if I can figure out what's going on.

I don't use a display manager or desktop environment, just EXWM via emacs as my window manager. Maybe there's some unmet dependency that everyone else is getting from their display manager?

Ambrevar commented:
I also use EXWM.

The only WM-related dependency is D-Bus, but since you can start Next
that's not your problem.

Next has an all-in-one Makefile that grabs Quicklisp and fetches the
deps for you.  You can type `make` and follow the instructions :)
Don't hesitate to come back to me if you need more help.

7c6f434c commented:
Built it in a no-DE StumpWM setup, wrapped it into a DBus session, seems to work from the first glance.

```
env -i DISPLAY=$DISPLAY $(nix-build /home/repos/nixpkgs-test -A nix)/bin/nix run -f /home/repos/nixpkgs-test procps next dbus -c dbus-launch next
```

Ambrevar commented:
@matthuszagh 

Try running Next by exporting WEBKIT_FORCE_COMPOSITING_MODE=1 and the
WEBKIT_DISABLE_COMPOSITING_MODE=1 environment variables (separately).

matthuszagh commented:
WEBKIT_DISABLE_COMPOSITING_MODE=1 works! Thanks @Ambrevar

Ambrevar commented:
Something is wrong between the Nvidia driver and Next, but if Epiphany
can get it right, Next can too!

nlewo commented:
I think we can merge this PR since the issue encountered by @matthuszagh doesn't seems to be related to the packaging.
Moreover, we could hope this issue will be fixed on the next `next` release;)
WDYT?

Ambrevar commented:
Thanks!

And thank you a lot, @lewo, for working on this!
